lib/segment/src/segment.rs

```diff
@@ -293,9 +305,13 @@ impl SegmentEntry for Segment {
     fn payload(
         &self,
         point_id: PointIdType,
+        filter_payload: Option<FilterPayload>,
```

Comment:
Modification of this function involves a lot of changes. I would prefer to have a separate function for this.

Reply:
I would even maybe moved filter function into `FilterPayload` so the call would look like `filter_payload.process(&payload) // returns new filtered payload`